### PR TITLE
retroarch.cfg: Reorganize sections

### PR DESCRIPTION
**Pull request by Stevoisiak**
Reorganizes sections in `retroarch.cfg` to match RGUI.

- Add new sections (Directory, Driver)
- Move settings to correct sections
- Add missing settings (`camera_driver`, `recording_output_directory`, etc)
- Update code comments

### CODE REVIEW

**Comment by Stevoisiak** [comment]
There are a few other sections in RGUI that are missing in `retroarch.cfg`. (Directory, Privacy, Achievements, etc).

Would there be interest in additional Pull Requests to add these sections?

File: retroarch.cfg [hunk 1]
```diff
@@ -138,6 +96,18 @@
 # By default, tries to use first suitable driver.
 # video_context_driver =
```
Comment (Stevoisiak) [inline retroarch.cfg]: Is the `video_context_driver` option accessible from RGUI?

Comment (Stevoisiak) [inline retroarch.cfg]: Is the `video_context_driver` option accessible from RGUI?

File: retroarch.cfg [hunk 2]
```diff
@@ -138,6 +96,27 @@
...
+# Default will use "sinc".
+# audio_resampler =
+
+# Camera driver.
+# camera_driver =
+
+# Location driver.
+# location_driver =
+
+# Menu driver to use. "rgui", "lakka", etc.
```
Comment (gouchi) [inline retroarch.cfg]: Also out of scope of this PR but you can take the opportunity to change from

`+#` Menu driver to use. "rgui", "lakka", etc.
to
`+#` Menu driver to use. "rgui", "xmb", "glui", etc.

Thank you.

Comment (Stevoisiak) [reply retroarch.cfg]: Should "etc" be removed if all three menu drivers are listed?

File: retroarch.cfg [hunk 3]
```diff
@@ -128,7 +128,13 @@
 # Shows hidden files and folders in directory listings.
 # show_hidden_files = false
 
-#### Video
+#### Driver
+
+# Input driver. Depending on video driver, it might force a different input driver.
+# input_driver = sdl
+
+# Joypad driver. (Valid: linuxraw, sdl, dinput)
```
[the remark this answers is not in the document]
Comment (Stevoisiak) [reply retroarch.cfg]: Was `sdl` removed? I don't see it as an option when running on a Raspberry Pi. (`dinput` is Windows exclusive)

File: retroarch.cfg (hunk 1 above)
Comment (twinaphex) [reply retroarch.cfg]: Nope it isn't. I would leave that out for now.

Comment (Stevoisiak) [reply retroarch.cfg]: Should I move it back to `Video`, or should it go in `Misc`?

File: retroarch.cfg [hunk 4]
```diff
@@ -275,10 +262,6 @@ video_message_bgcolor_green = 0
 video_message_bgcolor_blue = 0
 video_message_bgcolor_opacity = 1.0
```
[the remark this answers is not in the document]
Comment (RobLoach) [reply retroarch.cfg]: Yes, and I believe they are now...
```
video_msg_bgcolor_opacity
video_msg_bgcolor_red
video_msg_bgcolor_green
video_msg_bgcolor_blue
```

File: retroarch.cfg (hunk 2 above)
Comment (gouchi) [reply retroarch.cfg]: There are other menu drivers which are not enabled by default.

File: retroarch.cfg (hunk 4 above)
[the remark this answers is not in the document]
Comment (Stevoisiak) [reply retroarch.cfg]: @RobLoach You are correct. I'll change them with a commit

https://github.com/libretro/RetroArch/blob/master/configuration.h#L352-L354

File: retroarch.cfg [hunk 5]
```diff
@@ -261,23 +248,20 @@
...
+# video_msg_pos_x = 0.05
+# video_msg_pos_y = 0.05
 
-# Color for message. The value is treated as a hexadecimal value.
-# It is a regular RGB hex number, i.e. red is "ff0000".
-# video_message_color = ffffff
+# Color for message.
+# video_msg_color_r =
+# video_msg_color_g =
+# video_msg_color_b =
```
Comment (Stevoisiak) [inline retroarch.cfg]: What are the default values for `video_msg_color`?

Comment (Stevoisiak) [reply retroarch.cfg]: @RobLoach I intentionally changed it to `video_msg_color`. [According to configuration.h](https://github.com/libretro/RetroArch/blob/master/configuration.h#L274-L279), the variable names were changed at some point.

Comment (Stevoisiak) [reply retroarch.cfg]: @RobLoach I'll revert the variable names for now, since they could be considered outside the scope of my PR.

**Comment by Stevoisiak** [comment]
I've been a bit busy to work on this lately. I might come back to do more later, but this should be good to merge as-is.

**Comment by Stevoisiak** [comment]
@RobLoach Does anything else need to be done before this is merged?

**Comment by RobLoach** [comment]
@Stevoisiak Don't think so. Seems good to me!

@twinaphex Thoughts?

**Pull request merged**